frontend/src/vault/keysign/shared/utils/parseTransfferedKeysignMsg.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  KeysignMessage,
+  KeysignPayload,
+} from '../../../../gen/vultisig/keysign/v1/keysign_message_pb';
+import { keygenServerUrl } from '../../../keygen/KeygenServerType';
+import { decompressQrPayload } from '../../../qr/upload/utils/decompressQrPayload';
+import { getPayloadFromServer } from '../../../server/utils/getPayloadFromServer';
+
+export const parseTransferredKeysignMsg = async (
+  binary: Uint8Array
+): Promise<KeysignMessage> => {
+  const keysignMsg = KeysignMessage.fromBinary(binary);
+
+  if (keysignMsg.payloadId) {
+    const serverType = keysignMsg.useVultisigRelay ? 'relay' : 'local';
+    const serverUrl = keygenServerUrl[serverType];
```

Comment:
@rcoderdev ,I am afraid this one is not correct , when it is in "local" mode , when the initiate device set network mode to "local" , then the pair device will need to go through a service discovery to find out the relay server URL . 
Which is implemented in `KeysignServerUrlProvider` ,after that we can download the payload. 

Obviously , it is much easier to use "relay" server , as the relay server is static , it is there all the time